worker-markup-core/src/main/java/com/github/cafdataprocessing/worker/markup/core/FieldNameMapper.java

```diff
@@ -98,4 +101,48 @@ private static void renameField(Multimap<String, ReferencedData> mapData, Map.En
             mapData.removeAll(key);
         }
     }
+
+    private static void renameField(final Multimap<String, ReferencedData> mapData, final InputFieldMapping inputFieldMapping,
+                                    final DataSource dataSource)
+    {
+        LOG.trace("Trying to rename field {} to {}", inputFieldMapping.inputField, inputFieldMapping.mapToField);
+        if (mapData.containsKey(inputFieldMapping.inputField)) {
+            LOG.trace("Field found, renaming field {} to {}", inputFieldMapping.inputField, inputFieldMapping.mapToField);
+            // For each value associated with the input field, put the value into the new CAF field name
+            for(ReferencedData mapDataValue: mapData.get(inputFieldMapping.inputField))
+            {
+                if(inputFieldMapping.transform!=null)
+                {
+                    ReferencedData transformedValue;
+                    switch (inputFieldMapping.transform)
+                    {
+                        case epochSecondsToISO8601:
+                            transformedValue = epochSecondsToISO8601Transform(mapDataValue, dataSource);
+                            break;
+                        default:
+                            throw new RuntimeException("Unrecognized transform specified on input field mapping: "
+                                    +inputFieldMapping.transform.toString());
+                    }
+                    mapData.put(inputFieldMapping.mapToField, transformedValue);
+                }
+                else
+                {
+                    mapData.put(inputFieldMapping.mapToField, mapDataValue);
+                }
+            }
+            // Remove the instances of the input field in its original form
+            mapData.removeAll(inputFieldMapping.inputField);
+        }
+    }
+
+    private static ReferencedData epochSecondsToISO8601Transform(ReferencedData inputValue, DataSource dataSource)
+    {
+        LOG.debug("Retrieving epoch seconds value to transform to ISO8601");
+        String valueToTransform = ReferencedDataRetrieval.getContentAsStringEx(dataSource, inputValue);
+        LOG.debug("Parsing epoch seconds value to long.");
+        long input = Long.parseLong(valueToTransform);
```

Comment:
We need to catch the `NumberFormatException` here in case the value is not a number.  I'm OK with just re-throwing an exception but the message should state which field couldn't be transformed because it unexpectedly wasn't a number.